### ISSUE TITLE
file truncated when adding to .zip

### ISSUE THREAD

**Issue by michaeladamkatz** [pocoproject/poco#851]
Using poco 1.6.1 on Windows 7.

I am trying to create a .zip file that contains two files. One of the files is a .kml text (.xml) file, and when it gets added to the .zip, it gets truncated.

The relevant code is below. I tried using both the addRecursive() call (commented in below) and multiple addFile() calls (commented out below). In both cases the .kml file gets truncated in the same way.

```
    // File directory( p.scratchDirectoryPath );
    Path dstPath( p.exportDirectoryPath );
    dstPath.append( p.exportFileName );
    ofstream ofs( dstPath.toString().c_str(), ios::binary );
    Poco::Zip::Compress c( ofs, true );
    c.addRecursive( Path( p.scratchDirectoryPath ) );
    /*
    vector<string> files;
    directory.list( files );
    for ( vector<string>::iterator i = files.begin(); i != files.end() ; i++ )
    {
        Path srcPath( p.scratchDirectoryPath );
        srcPath.append( *i );
        c.addFile( srcPath, *i );
    }
    */
    c.close();
```

The relevant files can be found here:

https://github.com/michaeladamkatz/misc/tree/master/poco_zip_bug

The two files in the __scratch directory are the ones that are going into the .zip. The zipped file created is my_export.kmz. The file my_export.zip is is this same archive, just renamed to have a .zip ending. The file my_export.kml on the outer level is extracted from this .zip. As you can see, it is 4,257 bytes long, whereas the original file inside the __scratch directory is 7,450 bytes long. I don't know why it's getting cut off at 4,257 bytes. Perhaps this represents 1k of compressed data and it's hitting some buffer limit?

**Comment by michaeladamkatz** [comment]
In case it matters, I'm building and running with Visual Studio 2008 Express.

**Comment by ptarabbia** [comment]
Is this bug still happening with 1.6.1 ? I have a product shipping soon that uses the Zip features extensively for user-saved data, such a bug would be really critical, so I'm very interested to know if it's been fixed.

**Comment by michaeladamkatz** [comment]
I am using poco 1.6.1. I can't tell you what makes the problem happen, if it's a file of a certain length, or if it's something else particular to the data I'm zipping. (You can download my data files above and try it with code similar to mine above and see if you get the same problem. I would be interested to know, and it might give you an idea of whether you are likely to run into the same problem.)

However, I was also running into bugs with some of the poco archive unzip functionality, so I went to https://code.google.com/p/miniz/ for zip functionality and so far it seems to work well. (I still use poco for its other functionality.) Very easy to add and use.

**Comment by ptarabbia** [comment]
Yes, I had issues in 1.6.0 with corrupted CRC when the packed size is exactly 1024 - this was fixed by updating to 1.6.1, but if there more ZIP issues, I might stay with 1.6.0 and use miniz.
